core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -1242,6 +1244,37 @@ public Frame decorrelateRel(Join rel) {
     return register(rel, newJoin, mapOldToNewOutputs, corDefOutputs);
   }
 
+  public Frame decorrelateRel(LogicalUnion rel) {
+    return decorrelateRel((Union) rel);
+  }
```

Comment:
This is not needed, which can be covered by Union. Other operator has method for logical operator just for legacy reasons.

Reply:
Thanks ~ I will update soon.